### DISCUSSION TITLE
Two different ESP32 boards with OV5640 af camera

### DISCUSSION THREAD

**Discussion by Cysign** [arendst/Tasmota#22505] [Support request]
Hi,
I just got two different ESP32-WROVER-DEV module with OV5640 autofocus module. 

1. One looks like https://templates.blakadder.com/OV5640_V1_2.html
https://de.aliexpress.com/item/1005007884367777.html?spm=a2g0o.order_list.order_list_main.47.6f785c5fkitwoH&gatewayAdapt=glo2deu

But it is rev1.2.1 - no clue what the difference is.
I have Tasmota installed and the device connects to Wifi properly.
But I can't read out a BME280 using pins 21/SDL and 19/SCL (looks like pin 22 is not exposed on this board - probably any issues with using i2c on pin 19?).

2. The other board is this one:
https://de.aliexpress.com/item/1005006705436186.html?spm=a2g0o.order_list.order_list_main.5.6f785c5fkitwoH&gatewayAdapt=glo2deu

I flashed Tasmota and I can read out a BME280 using pins 21/SDA and 22/SCL.

3. I'd like to have a camera preview when I enter the IP address of the devices. I've done this with an ESP32-Cam module - but I cannot remember exactly what I did...

**Comment by Cysign** [comment]
Reading out the BME280 works on both devices. Just soldered the sensor to the wrong pin...

**Comment by Cysign** [comment]
Just flashed "[Tasmota 14.3.0 (release-webcam) by Theo Arends](https://bit.ly/tasmota)" on the USB-C device. Still couldn't get any camera preview. Also there is no sensor data anymore, though I set GPIO 16 & 17 for i2c (like on the tasmota image before).

**Reply by sfromis** [reply]
The webcam build does not include sensors, the list from command `I2CDriver` is empty.

**Reply by Cysign** [reply]
Thanks for the reply. Is there anything I can do about this?

Using the webcam-image I finally could get the USB-C version of the OV5640 running. Until I selected the I2C ports. Now I'm getting no camera preview anymore...
Also I wonder why there are two selections of the module available. configuration->module and configutarion->template.

**Reply by sfromis** [reply]
Templates is the newer and much improved feature for having a configuration template for many gpios in one step, as you can copy/paste a template string. Enabled by selecting module 0.

To include webcam and sensors in the same "original ESP32" binary, you can create a custom build.

**Reply by Cysign** [reply]
I tried to use one of the "more easy" toolchains - but I couldn't set them up. So I'm going to compile it using Platform IO.
